ir/visitor.cpp

```diff
@@ -416,3 +417,49 @@ ControlFlowVisitor &ControlFlowVisitor::flow_clone() {
     assert(rv->check_clone(this));
     return *rv;
 }
+
+#define DEFINE_APPLY_FUNCTIONS(CLASS, TEMPLATE, TT)                                             \
+    TEMPLATE bool IR::CLASS TT::apply_visitor_preorder(Modifier &v)                      \
```

Comment:
I don't like duplicating this big macro in both the header and the .cpp file -- need to define it once in the header and NOT `#undef` it (with a comment about leaving it valid so it can be expanded in visitor.cpp).  This does pollute the global namespace a bit, but that is unavoidable.